### PR TITLE
GTM support

### PR DESCRIPTION
**Pull request by dmitrijs-voronovs**
improvement of https://github.com/scandipwa/base-theme/pull/527

---
Integrated GTM into SWPWA according to this spreadsheet https://docs.google.com/spreadsheets/d/16Z86QjTXTQ5dLUH6yoi33y2ke-VQtFvmEgqoBw1iZl8/edit?usp=sharing

Depends on:
*  https://github.com/dmitrijs-voronovs/GTM
* https://github.com/dmitrijs-voronovs/GTM-graphql 

 How to test:
1. Register at [tagmanager.google.com](tagmanager.google.com) by mentioning the link of target website.
2. Get GTM_ID, that looks like `GTM-XXXXXXX`
3. In Magento Admin go to `Stores > Configuration > Scandiweb > PWA Google Tag Manager`
4. Insert GTM_ID and enable GTM as follows
image
5. On GTM website next to the GTM_ID click on preview button 
image
6. Refresh the target website.
7. Track all events
Selection_185

### CODE REVIEW

File: src/app/component/ProductCard/ProductCard.component.js [hunk 1]
```diff
@@ -55,6 +58,22 @@ export default class ProductCard extends PureComponent {
 
     imageRef = createRef();
 
+    handleClick = () => {
```
Comment (RoboLV) [inline src/app/component/ProductCard/ProductCard.component.js:61]: Shange it to simple method, as it rise problem in method rewriting from child theme

File: src/app/query/Config.query.js [hunk 2]
```diff
@@ -22,6 +22,19 @@ export class ConfigQuery {
             .addFieldList(this._getCheckoutAgreementFields());
     }
 
+    getGTMConfiguration() {
+        return new Field('getGtm')
+            .setAlias('gtm')
+            .addFieldList(this._getGTMConfigurationFileds());
+    }
+
+    _getGTMConfigurationFileds() {
```
Comment (RoboLV) [inline src/app/query/Config.query.js:31]: TYPO: _getGTMConfigurationFields

File: src/app/store/MyAccount/MyAccount.dispatcher.js [hunk 3]
```diff
@@ -47,11 +55,35 @@ export class MyAccountDispatcher {
...
             error => dispatch(showNotification('error', error[0].message))
         );
     }
 
+    /**
+     * transfer grouped products data from guest to logged in user
+     *
+     * @param {numbre} id customer id
+     */
+    transferGroupeProductsData(id) {
```
Comment (RoboLV) [inline src/app/store/MyAccount/MyAccount.dispatcher.js:70]: TYPO: transferGroupedProductsData

File: src/app/route/ProductPage/ProductPage.component.js [hunk 4]
```diff
@@ -35,6 +39,37 @@ export default class ProductPage extends PureComponent {
...
+        if (areDetailsLoaded) {
+            this._gtmProductDetail();
+        }
+    }
+
+    componentDidUpdate(prevProps) {
+        const { areDetailsLoaded, location: { pathname } } = this.props;
+        const { areDetailsLoaded: prevAreDetailsLoaded, location: { pathname: prevPathname } } = prevProps;
+
+        if (
```
Comment (RoboLV) [inline src/app/route/ProductPage/ProductPage.component.js:54]: No space needed

**Comment by alfredsgenkins** [comment]
Closed, @yeegor reworked into extension for 3.0.0

**Comment by yeegor** [comment]
This pull request is no longer relevant, in order to acquire. the GTM functionality feel welcome to use the extension for ScandiPWA v3 which may be [found on the marketplace](https://marketplace.scandipwa.com/google-tag-manager.html).

**Pull request closed without merging**